Api/Services/OrderService.cs

```diff
@@ -272,7 +278,17 @@ public async Task<Result<OrderVM>> UpdateOrderStatusAsync(int id, UpdateOrderSta
             };
         }
 
-        //is it in the correct restaurant group?
+        if (order.OrderItems.Any(item => item.Status == OrderStatus.Taken || item.Status == OrderStatus.Cancelled))
```

Comment:
Tu miałoby być sprawdzenie czy wszystkie itemy są `Taken` lub `Cancelled`. Bo inaczej jak tylko ustawimy  status jedenej pozycji na `Taken` lub `Cancelled`, to już nic nie zrobimy z tym orderem